src/components/Button.tsx

```diff
@@ -0,0 +1,79 @@
+import {StyleSheet, Text, TouchableOpacity} from "react-native";
+import React from "react";
+import {Ionicons} from "@expo/vector-icons";
+
+interface ButtonProps {
+    pressHandler?: () => void,
+    text?: string,
+    icon?: string,
+    styles?: {
+        button?: Record<string, any>,
+        text?: Record<string, any>,
+        icon?: Record<string, any>
+    }
+    type?: ButtonType,
+    size?: ButtonSize
+}
+
+export enum ButtonType {
+    primary,
+    secondary,
+    clean,
+    transparent
+}
+
+export enum ButtonSize {
+    sm,
+    md,
+    xl
+}
+
+const Button = (props: ButtonProps) => {
+    const {
+        pressHandler = () => {
+        },
+        text,
+        icon,
+        styles,
+        type,
+        size
+    } = props;
```

Comment:
Destruct props directly in the first row
```
const Button = ({ pressHandler, text, icon, etc}: ButtonProps) => 
```

Reply:
Sorry, but, in my humble opinion, this way is better because:
1. It's  easier to debug (i don't mean console.log "debug" ;) )
2. it's easier to read. 
So I'd prefer to leave it as is. :) But I will change it surely if you would be so kind to explain me why it's "smells" :)

Reply:
It's not really `bad smell`, it's more like a redundant line that increases the code amount. If you have just a stupid component, there is a difference
```
const Component = ({ test1, test2 }: Props) => <View /> 
```

```
const Component = (props: Props) => {
  const { test1, test2 } = props;
  return <View />
}
```

Reply:
But it's your project and it doesn't really affect the readability that much, so up to u :)

Reply:
Yeah, thanks! That's mainly because I am on the project where I often should debug a lot of information which goes through props. And it's easier to set breakpoints and check my whole `props`. :)